### PR TITLE
#19934 Fix error when create folder

### PR DESCRIPTION
**Pull request by freddyucv**
After this change 

https://github.com/dotCMS/core/commit/31f5c3f4dc542cc7591bb843a5fa2d9b37c7964c#diff-3680fcb1383b2f20c967e184d379c1c53a4fda303088b2c86573c83d6aa615daR278

the method doesUserHavePermission return true when a Admin user ask by permission in a new Permissionable, an then here

https://github.com/dotCMS/core/blob/f695ec4520a33400498e5c308f9b3ca9e8b102a3/dotCMS/src/main/webapp/html/portlet/ext/folders/edit_folder.jsp#L245

return true and show the permission tab for a new folder, when try to show this tab get a NullPointerException here:

https://github.com/dotCMS/core/blob/d15c5db8f08ed0148f9c33287cf1eefde11b45bc/dotCMS/src/main/webapp/html/portlet/ext/common/edit_permissions_tab_inc.jsp#L17

https://github.com/dotCMS/core/blob/5cfc08b6315a3b2f1ae71dcc43ccefbf1beb9740/dotCMS/src/main/java/com/dotmarketing/business/IdentifierAPIImpl.java#L92

### CODE REVIEW

**Comment by freddyucv** [comment]
> Code does not look like it is formatted properly - we use spaces rather than tabs. Please make sure you have the google java style guide installed:
> 
> https://github.com/dotCMS/core/tree/master/docs/styleguide

done https://github.com/dotCMS/core/pull/20025/commits/1ae102ec7e905bb9011fc5ea25548176161e35ac

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/business/PermissionBitAPITest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.exception.DotDataException;
+import com.liferay.portal.model.User;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+
+public class PermissionBitAPITest extends IntegrationTestBase  {
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/business/PermissionBitAPITest.java:16]: This class needs to be added to the MainSuite in order to be run

**Review by dsilvam — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java [hunk 2]
```diff
@@ -270,26 +270,31 @@ private boolean userPermissions(final UserProxy userProxy, final User userIn) {
...
-        if(UtilMethods.isEmpty(permissionable.getPermissionId())){
-            Logger.debug(this.getClass(), "Trying to get permissions on null inode of type :" + permissionable.getPermissionType()) ;
-            Logger.debug(this.getClass(), "Trying to get permissions on null inode of class :" + permissionable.getClass()) ;
-            return false;
-            
-        }
+
+		if (permissionable == null) {
+			Logger.warn(this, "Permissionable object is null");
+			throw new NullPointerException("Permissionable object is null");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java:281]: I think this should be Illegal Argument

File: dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java [hunk 3]
```diff
@@ -270,26 +270,31 @@ private boolean userPermissions(final UserProxy userProxy, final User userIn) {
...
+		}
+
+		if (UtilMethods.isEmpty(permissionable.getPermissionId())) {
+			Logger.debug(this.getClass(),
+					"Trying to get permissions on null inode of type :" + permissionable
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java:286]: Remember to use Logger.debug(this.getClass(), ()-> "....") version

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/business/PermissionBitAPITest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/business/PermissionBitAPITest.java:16]: done https://github.com/dotCMS/core/pull/20025/commits/4f24ed8150d8e42c574b1f93351d0d0e5813d9bc#diff-8c127e4dadcd12d480a62cfc2a0dd406dbcb61bab2081608e05dfd4be0df5499R392

File: dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java]: done https://github.com/dotCMS/core/pull/20025/commits/4f24ed8150d8e42c574b1f93351d0d0e5813d9bc#diff-3680fcb1383b2f20c967e184d379c1c53a4fda303088b2c86573c83d6aa615daR286

**Review by dsilvam — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]
Looks good, let's wait for the tests to merge

**Pull request closed without merging**